hwy/contrib/sort/sort_test.cc

```diff
@@ -618,6 +618,40 @@ std::vector<Algo> AlgoForTest() {
   };
 }
 
+template <class Traits>
+void TestPartialSort() {
+  constexpr size_t RANGE = 182;
+  using LaneType = typename Traits::LaneType;
+  using KeyType = typename Traits::KeyType;
+  const SortTag<LaneType> d;
+  detail::SharedTraits<Traits> st;
+  const Dist dist = Dist::kUniform16;
+
+  constexpr size_t kLPK = st.LanesPerKey();
+  // HWY_ALIGN LaneType
+  //     buf[SortConstants::BufBytes<LaneType, kLPK>(HWY_MAX_BYTES) /
+  //         sizeof(LaneType)];
+
+  const size_t log2 = 12;  // AdjustedLog2Reps(20);
+  const size_t num_lanes = 1ull << log2;
+  const size_t num_keys = num_lanes / kLPK;
+
+  auto aligned = hwy::AllocateAligned<LaneType>(num_lanes);
+  (void)GenerateInput(dist, aligned.get(), num_lanes);
+
+  std::vector<KeyType> s(aligned.get(), aligned.get() + num_lanes);
+
+  std::partial_sort(s.begin(), s.begin() + RANGE, s.end(),
+                    std::greater<KeyType>{});
+  PartialSort(d, st, aligned.get(), num_keys, RANGE);
+
+  // Select(d, st, aligned.get(), num_keys, buf, RANGE);
+  // std::sort(aligned.get(), aligned.get() + RANGE, std::greater<KeyType>{});
+
+  for (size_t i = 0; i < RANGE; ++i)
+    HWY_ASSERT(std::fabs(aligned.get()[i] - s[i]) < 0.01);
```

Comment:
Nit: 0.01f is more efficient (avoids promoting the expression to double).

0.01 is also pretty large epsilon because the random inputs are in [1, 2), perhaps 1E-4f?

Finally, vqsort supports f16. If you want to support that in this test as well, we can `ConvertScalarTo<float>` all of the sorted values before subtracting/comparing.